snakemake/hla.rules

```diff
@@ -0,0 +1,39 @@
+# Copyright (c) 2018. Mount Sinai School of Medicine
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+HLA typing and gene quantification.
+
+TODO: replace this with allele-specific quantification that works
+across multiple species.
+"""
+
+rule seq2hla_paired_end:
+  input:
+    r1 = "{prefix}_R1.fastq.gz",
+    r2 = "{prefix}_R2.fastq.gz",
+  output:
+    "{prefix}-ClassI.expression",
+    "{prefix}-ClassII.expression",
+    "{prefix}-ClassI.HLAgenotype4digits",
+    "{prefix}-ClassII.HLAgenotype4digits",
+  resources:
+    mem_mb = _mem_gb_for_alignment() * 1024
```

Comment:
it's confusing to use alignment-related resources in non-alignment jobs, so i'd say either use the `_mem_gb_for_ram_hungry_jobs` or make a new function specific to seq2hla. do you know how much memory it needs or are you just guessing?